Plugins/SwiftProtobufPlugin/plugin.swift

```diff
@@ -0,0 +1,139 @@
+import Foundation
+import PackagePlugin
+
+@main
+struct SwiftProtobufPlugin: BuildToolPlugin {
+    /// Errors thrown by the `SwiftProtobufPlugin`
+    enum PluginError: Error {
```

Comment:
Is this sufficient?  Do we need an error case for failure to find the protoc binary or for failure to run that binary?